content/en/blogs/devops-automatic-testing.md

```diff
@@ -0,0 +1,247 @@
+---
+title: 'Use KubeSphere DevOps to Build an Automated Testing System'
+tag: 'KubeSphere,Kubernetes'
+createTime: '2020-04-29'
+author: 'Shaowen Chen'
+snapshot: 'https://pek3b.qingstor.com/kubesphere-docs/png/20200428175258.png'
+---
+
+## Layers of Testing
+
+The purpose of testing is to verify expected functionality and find potential defects. Testing enhances the confidence in delivering qualified products and also brings the possibility of agile iteration. It is safe to say that testing determines the product development progress.
+
+The network model includes seven layers of OSI model and four layers of TCP model, and the development modes include MTV, MVC, MVP, MVVM, etc. These technologies gradually mature as they grow from high cohesion, low coupling, and division of responsibilities, modules, and layers to structuralization and standardization.
```

Comment:
->
As you may already know, in networking, there are 7 layers of OSI model and 4 layers of TCP model. In development, we also have different patterns such as MTV, MVC, MVP, and MVVM. These technologies have become more mature featuring high cohesion, low coupling, and division of responsibilities, modules, and layers. They have helped us build architectures and standards.